src/main/java/racingcar/domain/Car.java

```diff
@@ -1,46 +1,61 @@
 package racingcar.domain;
 
-public class Car {
-    static final int CRITERION_FOR_MOVING = 4;
-
-    private int location = 0;
-    private String name = "Unknown";
+public class Car implements Comparable<Car> {
+    private Location location;
+    private Name name;
 
     Car(String name) {
-        if (isValidName(name)) {
-            this.name = name;
-        }
+        this(name, 0);
     }
 
-    Car() {}
+    Car(String name, int location) {
+        this.name = new Name(name);
+        this.location = new Location(location);
+    }
 
-    public int getLocation() {
-        return location;
+    Car() {
+        this("Unknown");
     }
 
     void goForward() {
-        location++;
+        location = location.goForward();
     }
 
-    boolean isAbleToMove(int value) throws IllegalArgumentException {
-        if (value < 0 || value > 9) {
-            throw new IllegalArgumentException("invalid argument for isAbleToMove : " + value);
+    void move(MovingStrategy movingStrategy) {
+        if (movingStrategy.isMovable()) {
+            goForward();
         }
+    }
 
-        return value >= CRITERION_FOR_MOVING;
+    public String getNameInString() {
+        return this.name.toString();
     }
 
-    void move(int value) {
-        if (isAbleToMove(value)) {
-            goForward();
-        }
+    @Override
+    public int compareTo(Car o1) {
+        return location.compareTo(o1.location);
     }
 
-    public String getName() {
-        return this.name;
+    public String getCarStateInString() {
+        return name.toString() + " : " + location.toString();
     }
 
-    boolean isValidName(String name) {
-        return !(name == null || name.trim().isEmpty());
+    public boolean equals(Object o) {
+        if (o instanceof Car) {
+            Car compared = (Car) o;
+            return name.equals(compared.name) &&
+                    location.equals(compared.location);
+        }
+
+        return false;
+    }
+
+    public boolean equalsLocation(Object o) {
```

Comment:
```suggestion
    public boolean equalsLocation(Car car) {
```

car만 받을 수 있도록 작성한다면 조금 더 깔끔한 코드를 작성할 수 있을 것 같아요 :)